c7n/resources/elasticache.py

```diff
@@ -254,18 +289,24 @@ def _elasticache_snapshot_tags(
     """ Augment ElastiCache snapshots with their respective tags
     """
 
+    # added if statement to ensure snapshot is available in order to list tags
     def process_tags(snapshot):
         client = local_session(session_factory).client('elasticache')
         arn = generator(snapshot[model.id])
-        tag_list = retry(
-            client.list_tags_for_resource,
-            ResourceName=arn)['TagList']
-        snapshot['Tags'] = tag_list or []
-        return snapshot
+        if snapshot['SnapshotStatus'] == 'available':
+            tag_list = retry(
+                client.list_tags_for_resource,
+                ResourceName=arn)['TagList']
+            snapshot['Tags'] = tag_list or []
+            return snapshot
 
     with executor_factory(max_workers=2) as w:
         return list(w.map(process_tags, snapshots))
 
 
 def _cluster_eligible_for_snapshot(cluster):
-    return cluster['Engine'] != 'memcached'
+    # added regex search to filter unsupported cachenode types
+    return (
+        cluster['Engine'] != 'memcached' and not 
+        re.search('cache.t', cluster['CacheNodeType'])
```

Comment:
can you compile this as a top level module constant, avoids regex compilation per cluster.